game/kernel/asm_funcs_arm64.s

```diff
@@ -0,0 +1,262 @@
+;; GOAL Runtime assembly functions. These exist only in the arm64 version of GOAL.
+;; - https://developer.apple.com/documentation/xcode/writing-arm64-code-for-apple-platforms#Pass-arguments-to-functions-correctly
+;; - https://en.wikipedia.org/wiki/Calling_convention#ARM_(A64)
+;; - https://student.cs.uwaterloo.ca/~cs452/docs/rpi4b/aapcs64.pdf
+;; - s16–s31 (d8–d15, q4–q7) must be preserved
+;; - s0–s15 (d0–d7, q0–q3) and d16–d31 (q8–q15) do not need to be preserved
+;; - https://devblogs.microsoft.com/oldnewthing/20220728-00/?p=106912
+;; - ;; - https://courses.cs.washington.edu/courses/cse469/19wi/arm64.pdf
+
+.text
+
+;; Call C++ code on arm64 systems, from GOAL.
+;; Following the macOS documentation which mostly aligns with standard arm64
+.global _arg_call_arm64
+.align 4
+_arg_call_arm64:
+  ldr x8, [sp], #16
+
+  ; Putting an exclamation point after the close-bracket 
+  ; means that the calculated effective address is written back to the base register. (pre-indexing)
+  stp q15, q14, [sp, #-32]!
+  stp q13, q12, [sp, #-32]!
+  stp q11, q10, [sp, #-32]!
+  stp q9, q8, [sp, #-32]!
+
+  blr x8
+
+  ldp x8, x9, [sp], #32
+  ldp x10, x11, [sp], #32
+  ldp x12, x13, [sp], #32
+  ldp x14, x15, [sp], #32
```

Comment:
I think these should follow the same order as the stp: `ldp x15, x14...` (but it's been a while, i could be wrong)

Reply:
Nope you're right, I had to rewrite the stack pushing/popping too many times and got that wrong.